### PR TITLE
Aquifer refactor aquancon

### PR DESCRIPTION
**Pull request by joakim-hove**
Downstream of: https://github.com/OPM/opm-common/pull/1452

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this opm-common=1452 please

**Comment by joakim-hove** [comment]
@GitPaean : The intention was "just" to adapt to changes in opm-common; and to prepare for initiiazling the aquifer with deck access - but there was a somewhat significant refactor of the Aquifer implementation on the way. There are some details on the serialization which is still in progress

**Comment by GitPaean** [comment]
jenkins build this opm-common=1452 please

**Comment by GitPaean** [comment]
My local compilation failed. Let me check that first.

File: opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp [hunk 1]
```diff
@@ -158,20 +158,16 @@ BlackoilAquiferModel<TypeTag>::init()
         const AquiferCT aquiferct = AquiferCT(eclState.getTableManager(), deck);
         const Aquancon aquifer_connect = Aquancon(eclState.getInputGrid(), deck);
 
-        std::vector<AquiferCT::AQUCT_data> aquifersData = aquiferct.data();
-        std::vector<Aquancon::AquanconOutput> aquifer_connection = aquifer_connect.getAquOutput();
-
         assert(aquifersData.size() == aquifer_connection.size());
```
Comment (GitPaean) [inline opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp]: This assertion will not compile without `-DNDEBUG`.

File: opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp [hunk 2]
```diff
@@ -188,20 +184,16 @@ BlackoilAquiferModel<TypeTag>::init()
         const Aquifetp aquifetp = Aquifetp(deck);
         const Aquancon aquifer_connect = Aquancon(eclState.getInputGrid(), deck);
 
-        std::vector<Aquifetp::AQUFETP_data> aquifersData = aquifetp.data();
-        std::vector<Aquancon::AquanconOutput> aquifer_connection = aquifer_connect.getAquOutput();
-
         assert(aquifersData.size() == aquifer_connection.size());
```
Comment (GitPaean) [inline opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp]: Same here. `aquifersData` is not defined.

Comment (joakim-hove) [reply opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp]: fixed

File: opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp (hunk 1 above)
Comment (joakim-hove) [reply opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp]: fixed

**Comment by joakim-hove** [comment]
jenkins build this opm-common=1452 please

**Comment by GitPaean** [comment]
For the big case I am looking at, this PR or the upstream one introduces some small changes in the result. But the change is good. The results are slightly better and the running is slightly more robust and faster. 

It is possible it fixes some bugs from the old code. Not totally sure yet, will look into more.

**Comment by joakim-hove** [comment]
Update on this?

**Comment by GitPaean** [comment]
Hi, today I am out of the office due to private cause. I will take a look at this hopefully tomorrow.

**Comment by GitPaean** [comment]
@akva2 , maybe you can take a look at the ParallelRestart part .

**Comment by GitPaean** [comment]
For this PR, other than `ParallelRestart.cpp` related part, it looks good to me.  I am finalizing the reviewing of the upstream PR.

**Comment by akva2** [comment]
serialization looks fine apart from not respecting the local code style.

**Comment by joakim-hove** [comment]
> apart from not respecting the local code style.

I don't know exactly what you are aiming at here; I agree you could say that the naming does not follow local code style; it follows the relevant upstream - and that again follows the manual very closely. I can change it - but not really certain what would constitute *good* names for the various quantities.

**Comment by akva2** [comment]
All i meant was the brace placement :)

**Pull request merged**